### DISCUSSION TITLE
How add new failregex to exim-spam.conf

### DISCUSSION THREAD

**Discussion by ysha0** [fail2ban/fail2ban#3499] [General]
Hi, I'm trying to add new exim-spam.conf failregex based on lines from the log:
```
2023-04-20 04:31:30 H=(218bd1cb2febc55e7dd79e24f0341cb5eae1) [171.242.65.254] F=<qyhhevdooccok@bs-akustikbau.de> rejected RCPT <xxx@yyy.zz>: HELO locally blacklisted
2023-02-03 08:53:49 H=vps-9327f638.vps.ovh.net (51.178.51.28) [51.178.51.28] F=<dasdacfdf@gmail.com> rejected RCPT <xxx@yyy.zz>: Your IP in HELO
```
How look regular expression for using with known filter.d/exim-spam.conf failregex  for catch IP and "HELO locally blacklisted", "Your IP in HELO" like lines from log specified above?

**Comment by sebres** [comment]
filter [exim-spam](https://github.com/fail2ban/fail2ban/blob/master/config/filter.d/exim-spam.conf) already has similar line (but expected `dnsbl` in reason of rejection):
https://github.com/fail2ban/fail2ban/blob/2b98f461bb93c6a1a1cec1481fa94b7a84342977/config/filter.d/exim-spam.conf#L30

So you can add this:
```regex
# to catch all rejected
^\s*<host_info>F=(?:<>|[^@]+@\S+) rejected RCPT\b
# to catch only expected reasons:
^\s*<host_info>F=(?:<>|[^@]+@\S+) rejected RCPT [^@]+@\S+: (?:HELO locally blacklisted|Your IP in HELO)\b
```

Another very simple variant, without `<host_info>` tag (from exim common include), may look like this:
```
# to catch all rejected
^[^\[]*\[<ADDR>\] F=(?:<[^>]*>|[^@]+@\S+) rejected RCPT
# to catch only expected reasons:
^[^\[]*\[<ADDR>\] F=(?:<[^>]*>|[^@]+@\S+) rejected RCPT [^@]+@\S+: (?:HELO locally blacklisted|Your IP in HELO)\b
```
